enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,299 @@
+
+**Risk**: Bootstrap logic can overwrite other enabled features.
+
+*Mitigation*: Bootstrap logic is only invoked if the cluster uses the default feature set.
+
+**Risk:** Clusters need tech preview features AND the Jenkins pipeline build strategy enabled.
+
+*Mitigation*: Clusters with tech preview features are automatically marked as `NoUpgrade` clusters.
+If necessary, a cluster that needs tech preview features and the Jenkins pipeline build strategy can use the `CustomNoUpgrade` feature set,
+which gives admins the ability to enable individual feature gates on their own.
+
+**Risk:** Clusters need fully supported extra features AND the Jenkins pipeline build strategy enabled.
+
+*Mitigation*: There is no real mitigation for this other than using the `CustomNoUpgrade` feature set.
+The downside of this approach is that the cluster cannot be upgraded to the next minor release.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. How many clusters use Jenkins pipeline builds and also enable a supported feature gate?
+   None today, but that may change if other build related features require a feature gate (example - tech preview items).
+
+### Test Plan
+
+CI already has an existing test suite for the JenkinsPipeline build strategy.
+Setup for this suite will need to enable the `DeprecatedJenkinsPipeline` feature set before any tests run.
+A separate test will need to run against a standard OpenShift cluster that verifies JenkinsPipeline builds fail immediately.
+
+Tests for the `oc` command line will need to be updated as follows:
+
+* JenkinsPipeline builds started via `oc start-build` should issue a warning.
+* JenkinsPipeline BuildConfigs created via `oc new-build` should issue a warning.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Tech Preview -> GA
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Removing a deprecated feature
+
+The Jenkins pipeline strategy has been officially deprecated since OCP 4.1.
+This proposal does not permanently disable the strategy - rather it discourages its use by requiring admins to enable a feature set.
+This will be documented in all places where Jenkins is referenced.
+
+Eventually we would like to make this feature set a `NoUpgrade` feature set.
+Doing so would keep the Jenkins pipeline strategy enabled, but prevent the cluster from being upgraded to the next OCP minor version.
+Hypothetically this is possible:
+
+* In OCP 4.N the feature set is renamed from `DeprecatedJenkinsPipeline` to `DeprecatedJenkinsPipelineNoUpgrade`
+* openshift-controller-manager-operator for OCP 4.N automatically renames the configured feature set from `DeprecatedJenkinsPipeline` to `DeprecatedJenkinsPipelineNoUpgrade`.
+* Once the upgrade to 4.N is complete, the `*NoUpgrade` feature set automatically marks the cluster as not upgradeable.
```

Comment:
i don't think the idea of turning someone's prod cluster into a non-upgradeable cluster w/o them realizing that was going to happen.

I think there might be a better way to do this...... in 4.8/4.9 we are introducing the ability to define generic upgrade-gates that require admin acknowledgement.

If/when we have a 4.N that is going to remove jenkins, then in 4.N-1 we can add a required admin-ack gate that forces admins to acknowledge that the jenkins apis are being removed (and we can potentially only require that ack if there are buildconfigs that use jenkins).